src/Azure/Azure.Quantum.Client/SubmitterFactory.cs

```diff
@@ -62,8 +62,8 @@ public static class SubmitterFactory
         /// <param name="workspace">The workspace used to manage jobs.</param>
         /// <param name="storageConnection">The connection string for the storage account.</param>
         /// <returns>A QIR submitter.</returns>
-        public static IQirSubmitter? QirSubmitter(string target, IWorkspace workspace, string? storageConnection) =>
-            Submitter<IQirSubmitter>(QirSubmitters, target, workspace, storageConnection);
+        public static IQirSubmitter? QirSubmitter(string target, IWorkspace workspace, string? storageConnection, string targetCapability) =>
```

Comment:
I believe `targetCapability` shouldn't be part of these factory methods since that information is already part of the `SubmissionOptions` object.

Reply:
I take this back, we do need `targetCapability` in these methods.